google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/scrape/metadata_scraper.py

```diff
@@ -77,6 +87,96 @@ def _get_metadata_from_rdbms_connection(self, connection_args, query):
+                    base_dataframe, metadata_definition)
+            refresh_queries = query_assembler.get_refresh_metadata_queries(
+                exact_table_names)
+            logging.info('Refreshing metadata')
+            self._refresh_metadata_from_rdbms_connection(
+                connection_args, refresh_queries)
+        if user_config.scrape_optional_metadata:
+            optional_metadata = user_config.get_chosen_metadata_options()
+            optional_queries = query_assembler.get_optional_queries(
+                optional_metadata)
+            logging.info(
+                'Scraping metadata according to configuration file: {}'.format(
+                    optional_metadata))
+            enriched_dataframe = \
+                self._get_optional_metadata_from_rdbms_connection(
+                    connection_args, optional_queries, base_dataframe,
+                    metadata_definition)
+        return enriched_dataframe
+
+    def _refresh_metadata_from_rdbms_connection(self, connection_args,
+                                                refresh_queries):
+        con = None
+        try:
+            con = self._create_rdbms_connection(connection_args)
+            cur = con.cursor()
+            start_update = time.time()
+            for query in refresh_queries:
+                self._execute_refresh_query(cur, query)
+            end_update = time.time()
+            logging.info(
+                'Metadata analysis took {} seconds to run.'
+                'You can turn it off in ingest_cfg.yaml configuration file, '
+                'using refresh_metadata_tables flag'.format(end_update -
+                                                            start_update))
+        except:  # noqa:E722
+            logging.error(
+                'Error connecting to the database to update metadata.')
+            raise
+        finally:
+            if con:
+                con.close()
+
+    def _get_optional_metadata_from_rdbms_connection(self, connection_args,
+                                                     optional_queries,
+                                                     base_dataframe,
+                                                     metadata_definition):
+        con = None
+        try:
+            con = self._create_rdbms_connection(connection_args)
+            cur = con.cursor()
+            for query in optional_queries:
+                cur.execute(query)
+                rows = cur.fetchall()
+                if len(rows) == 0:
+                    warnings.warn(
+                        "Query {} delivered no rows. Skipping it.".format(
+                            query))
+                else:
+                    new_dataframe = self._create_dataframe(rows)
+                    new_dataframe.columns = [
+                        item[0].lower() for item in cur.description
+                    ]
+                    base_dataframe = self._get_merged_dataframe(
```

Comment:
rename this variable?

This contains a wrong meaning should be merged_dataframe, no?